### ISSUE TITLE
utoronto - prometheus gaps in data 

### ISSUE THREAD

**Issue by pnasrat** [2i2c-org/infrastructure#2634]
I was trying to do some explore queries of the hub for 500 but they weren't working

![image](https://github.com/2i2c-org/infrastructure/assets/13798/6f51ce48-772f-4118-8b74-aa217dc17c9c)

_Originally posted by @pnasrat in https://github.com/2i2c-org/infrastructure/issues/2628#issuecomment-1583660840_

**Comment by consideRatio** [comment]
> I was also trying to do some explore queries of the hub for 500 but they weren't working. This might be due to the RBAC change in the cluster?

I don't think this gap or issues are related to k8s RBAC specifically, but maybe the upgrade for some other reason. The utoronto k8s cluster upgrade initially failed, so a new cluster was created, but in the end we figured out how to get the old cluster to work with its new k8s version. So, utoronto is at k8s 1.26 using the old k8s cluster, without k8s RBAC enabled.

The actual gap is something i suspect related to the node-exporter failing to be scraped by the prometheus-server during this time for some reason.

The gap is observed 18 to 21h from now, and it seems that it was from the node running the daemonset pod `support-prometheus-node-exporter-n6szb`, which hasn't been restarted as far as k8s is aware.

```
support-prometheus-node-exporter-2v9n5              1/1     Running   0          2d4h
support-prometheus-node-exporter-49njz              1/1     Running   0          2d4h
support-prometheus-node-exporter-4wsjq              1/1     Running   0          6h27m
support-prometheus-node-exporter-5b4xp              1/1     Running   0          7h50m
support-prometheus-node-exporter-cqvdh              1/1     Running   0          2d4h
support-prometheus-node-exporter-n6szb              1/1     Running   0          2d4h
support-prometheus-node-exporter-qqmcp              1/1     Running   0          2d4h
support-prometheus-node-exporter-rrjvn              1/1     Running   0          2d4h
support-prometheus-server-8585d8678c-nk2k7          2/2     Running   0          2d3h
```

The logs of that metrics exporter repeats the following, with the last message being from a time matching the gap.

```
ts=2023-06-08T19:56:54.525Z caller=stdlib.go:105 level=error caller="error encoding and sending metric family: write tcp 10.1.0.15:9100" msg="->10.1.0.5:49060: write: broken pipe"
```

My conclusion is that that error related to the gap then, uncertain about more, but I can tell from other node exporter pods that they don't have such errors in them.

I also see no issues seems to show in prometheus-server's logs in that time, but I'm not sure I would see a failure to scrape from some place.

I suspect that something going on, on that user node, caused significant disruption in general on the entire node.

**Comment by consideRatio** [comment]
Looking at grafana dashboards, I see one user that does something during exactly that time.

![image](https://github.com/2i2c-org/infrastructure/assets/3837114/98e0d0e9-7429-43e0-b830-053e12cb5682)

Isolated, this user has been active like this:

![image](https://github.com/2i2c-org/infrastructure/assets/3837114/2cde6702-a131-415f-b53e-6c2faf775cf3)

**Comment by consideRatio** [comment]
I had some additional ideas:

1. Maybe some process inside the container is memory culled?
   It seems that cant explain it based on looking at memory culling via the "cluster dashboard" in grafana.
2. Maybe gaps are recently introduced, or goes far back in time?
   It seems to go back in time before the k8s cluster upgrade, and before [this version bump of prometheus chart](https://github.com/2i2c-org/infrastructure/pull/2593) from 21.1.2 -> 22.6.2, also before [this previous version bump of prometheus chart](https://github.com/2i2c-org/infrastructure/pull/2516) from 20.2.0 -> 21.1.2... Overall, not sure when this started, but I don't see it in 2022 I think, but I see it in 2023-03-22 for example.
   
### Related

- Version of node-exporter chart used by prometheus chart: https://github.com/prometheus-community/helm-charts/blob/c89a0b713af61f507c01a11e61015ed11a5ad4f0/charts/prometheus/Chart.yaml#L36-L39
- Maybe relevant issue: https://github.com/prometheus/node_exporter/issues/2387

**Comment by consideRatio** [comment]
I suggested an increase from the requested CPU from the node-exporter, from 5m to 10m, thinking it fails to respond fast enough and prometheus-server breaks the connection at that point.

**Comment by consideRatio** [comment]
I extracted https://github.com/2i2c-org/infrastructure/issues/2647 from this issue, where I first focused on node-exporter gaps.
